src/drivers/CacheDriver.ts

```diff
@@ -0,0 +1,42 @@
+import {inject, injectable} from 'inversify';
+import {getLogger} from '../logger';
+import CacheDriverInterface from '../interfaces/CacheDriverInterface';
+import {RedisConnection} from '../services/redis/RedisConnection';
+import {IHandyRedis} from 'handy-redis';
+import {PS2AlertsCensusCacheTypes} from '../interfaces/PS2AlertsCensusCacheTypes';
+
+@injectable()
+export default class CacheDriver implements CacheDriverInterface {
+    private static readonly logger = getLogger('CacheDriver');
+
+    private readonly cacheClient: IHandyRedis;
+
+    constructor(@inject(RedisConnection) cacheClient: RedisConnection) {
+        this.cacheClient = cacheClient.client;
+    }
+
+    public async exists(identifier: string, type: PS2AlertsCensusCacheTypes): Promise<number|null> {
```

Comment:
I would take this method out as it is only going to lead to race conditions.

Reply:
How so?

Reply:
Because of the asynchronous nature of js. I assume you want to get some data so you check if exists when it doesn't you want to retrieve the data you want to cache. Basically between checking and setting there is a moment you can check again, but the data is not set. So you want to retrieve it(for a second time)